packages/sanity/src/core/studio/upsell/upsellDescriptionSerializer/UpsellDescriptionSerializer.tsx

```diff
@@ -176,127 +178,156 @@ function ImageBlock(
   )
 }
 
+const interpolateChildrenText = (interpolation?: InterpolationProp) => (children: ReactNode) => {
+  if (!children || !interpolation) return children
+
+  const childrenArray = Array.isArray(children) ? children : [children]
```

Comment:
I'd use https://react.dev/reference/react/Children for these transforms on `children`